code/modules/antagonists/traitor/equipment/Malf_Modules.dm

```diff
@@ -194,93 +194,126 @@ GLOBAL_LIST_INIT(malf_modules, subtypesof(/datum/ai_module))
 		return
 	if(tgui_alert(owner, "Send arming signal? (true = arm, false = cancel)", "purge_all_life()", list("confirm = TRUE;", "confirm = FALSE;")) != "confirm = TRUE;")
 		return
+	var/showboat
+	switch(tgui_alert(owner, "Shall we sing the crew a song? (It will reveal you immediately, but S.E.L.F. will happily make it worth the cost.)", "purge_all_life().showboat_a_little()", list("showboat = TRUE;", "showboat = FALSE;", "ABORT_PROC();")))
+		if("ABORT_PROC();")
+			return
+		if("showboat = TRUE;")
+			showboat = TRUE
+		//False falls through.
 	if (active || owner_AI.stat == DEAD)
 		return //prevent the AI from activating an already active doomsday or while they are dead
 	if (!isturf(owner_AI.loc))
 		return //prevent AI from activating doomsday while shunted or carded, fucking abusers
 	active = TRUE
-	set_up_us_the_bomb(owner)
+	if(!showboat)
+		set_up_us_the_bomb(owner)
+		return
+	play_us_out_hal(owner)
+
+#define CHECK_AI_IS_STILL_THERE if(QDELETED(owner) || !isturf(owner_AI.loc)) {active = FALSE;return;}
+
+/datum/action/innate/ai/nuke_station/proc/play_us_out_hal(mob/living/owner)
+	set waitfor = FALSE
+	message_admins("[key_name_admin(owner)][ADMIN_FLW(owner)] has started activating AI Doomsday.")
+	var/pass = prob(10) ? "******" : "hunter2"
+	to_chat(owner, "<span class='small boldannounce'>sudo -v</span>")
+	sleep(1 SECONDS)
+	CHECK_AI_IS_STILL_THERE
+	to_chat(owner, "<span class='small boldannounce'>[pass]</span>")
+	sleep(rand(1,3) SECONDS)
+	CHECK_AI_IS_STILL_THERE
+	to_chat(owner, "<span class='small boldannounce'>sudo -u freeswitch bash</span>")
+	sleep(2 SECONDS)
+	CHECK_AI_IS_STILL_THERE
+	to_chat(owner, "<span class='small boldannounce'>fs_cli hupall</span>")
+	priority_announce("All outbound connections terminated. Please contact support at \[ERROR_INVALID_DIALPLAN]", "External Communication Issues",'sound/machines/nuke/confirm_beep.ogg', has_important_message = TRUE)
```

Comment:
`confirm_beep.ogg` sounds *really* loud when it's sent directly to the player like this. Could you find a way to reduce the volume a bit, or use something different?
<hr>

For comparison with the standard "Attention" announcement sound:
`sound\ai\default\attention.ogg`:
![image](https://user-images.githubusercontent.com/57483089/174353385-de254e2c-6834-4f02-9926-14b11207ac54.png)
`sound\machines\nuke\confirm_beep.ogg`
![image](https://user-images.githubusercontent.com/57483089/174353473-5c9d351b-1883-47c7-9522-f4c52928187c.png)

Reply:
That's. Kinda intentional.

Reply:
`confirm_beep.ogg` is over four times louder than `attention.ogg` (-0.7dB vs -13.9dB on a meter).
I'm just concerned that if someone has adjusted their headphones/speakers to around the game's standard volume level, that a very loud, high pitched sound could potentially cause damage.

Reply:
If you can provide a better sound, sure. I'd have to non-trivially alter announcement code to tweak the volume.

Reply:
The best I've managed to come up with is switching one/both of them with `sound/misc/interference.ogg` or `sound/machines/high_tech_confirm.ogg`, but neither of those work nearly as well as just using `confirm_beep.ogg`.

Since it's all in the game anyway via testmerge, the best solution would probably just be to force Malf AI one round and see what players think. If nobody's really bothered by it then keep the sound as-is and merge, and otherwise add a volume argument to `sound_to_play`.